plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -643,6 +651,36 @@ private boolean deleteObject(String key)
         }
     }
 
+    @Override
+    public void deleteFiles(Collection<Path> paths)
+            throws IOException
+    {
+        try {
+            Iterator<List<Path>> partitions = Iterables.partition(paths, BATCH_DELETE_PARTITION_SIZE).iterator();
+            while (partitions.hasNext()) {
```

Comment:
`Iterables.partition` probably returns an Iterable, so you can use a `for` enumeration loop